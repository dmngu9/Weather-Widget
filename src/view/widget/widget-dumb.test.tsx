import * as React from 'react';
import { shallow } from 'enzyme';

import WidgetDumb from './widget-dumb';

const TITLE = 'Forecast of the day';
const CITY = 'Sydney';
const TEMPERATURE = 34;
const WIND_SPEED = '100km/h';
const WIND_DIRECTION = 'NE';
const ICON_URL = 'url';

describe('<WidgetDumb />', () => {
    it('should render correctly when error', () => {
        const wrapper = shallow(
            <WidgetDumb
                title={TITLE}
                showWind
                error="this is error"
                city={CITY}
                temperature={TEMPERATURE}
                windSpeed={WIND_SPEED}
                windDirection={WIND_DIRECTION}
                iconUrl={ICON_URL}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when loading', () => {
        const wrapper = shallow(
            <WidgetDumb
                title={TITLE}
                showWind
                loading
                city={CITY}
                temperature={TEMPERATURE}
                windSpeed={WIND_SPEED}
                windDirection={WIND_DIRECTION}
                iconUrl={ICON_URL}
            />
        );

        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <WidgetDumb
                title={TITLE}
                showWind
                city={CITY}
                temperature={TEMPERATURE}
                windSpeed={WIND_SPEED}
                windDirection={WIND_DIRECTION}
                iconUrl={ICON_URL}
            />
        );

        expect(wrapper.dive().dive()).toMatchSnapshot();
    });

    it('should render correctly with no wind data displayed', () => {
        const wrapper = shallow(
            <WidgetDumb
                title={TITLE}
                showWind={false}
                city={CITY}
                temperature={TEMPERATURE}
                windSpeed={WIND_SPEED}
                windDirection={WIND_DIRECTION}
                iconUrl={ICON_URL}
            />
        );

        expect(wrapper.dive().dive()).toMatchSnapshot();
    });
});
