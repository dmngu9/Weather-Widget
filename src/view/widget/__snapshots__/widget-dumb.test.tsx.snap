// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WidgetDumb /> should render correctly 1`] = `
<styled.div>
  <styled.h3>
    Forecast of the day
  </styled.h3>
  <styled.div>
    <styled.div>
      <styled.img
        alt="weather icon"
        src="url"
      />
    </styled.div>
    <styled.div>
      <styled.p>
        Sydney
      </styled.p>
      <styled.p>
        34
        °
      </styled.p>
      <styled.p>
        <strong>
          Wind
        </strong>
         
        NE
         
        100km/h
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`<WidgetDumb /> should render correctly when error 1`] = `
<styled.p>
  this is error
</styled.p>
`;

exports[`<WidgetDumb /> should render correctly when loading 1`] = `
<Spinner
  delay={100}
  invertColor={false}
  isCompleting={false}
  onComplete={[Function]}
  size="large"
/>
`;

exports[`<WidgetDumb /> should render correctly with no wind data displayed 1`] = `
<styled.div>
  <styled.h3>
    Forecast of the day
  </styled.h3>
  <styled.div>
    <styled.div>
      <styled.img
        alt="weather icon"
        src="url"
      />
    </styled.div>
    <styled.div>
      <styled.p>
        Sydney
      </styled.p>
      <styled.p>
        34
        °
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;
